### ISSUE TITLE
Semi-automatic adaptive font size

### ISSUE THREAD

**Issue by felixbraun97** [microsoft/vscode#97376]
TLDR: Please add additional **semi-automatic adaptive** options to specify the **font size**. 
Simpler alternative: Please add option to bind font size levels as a shortcut.

1) This option should be possible to be enabled or disabled via a keyboard shortcut. The default font size should be used as fallback.
2) Select font size (semi-)automatically depending on the **window size**. 
The user should be able to specify a largest and a smallest font for this. 
The editor should then always select the font size such that as much of the code can be displayed as possible whilest still being within the specified **font size range**.

**Comment by alexdima** [comment]
Here are my keybindings for this:

```
	// Use font zooming and never window zooming
	{ "key": "cmd+numpad_add",		"command": "editor.action.fontZoomIn", "when": "isMac" },
	{ "key": "shift+cmd+=",			"command": "editor.action.fontZoomIn", "when": "isMac" },
	{ "key": "cmd+=",				"command": "editor.action.fontZoomIn", "when": "isMac" },
	{ "key": "cmd+numpad_subtract",	"command": "editor.action.fontZoomOut", "when": "isMac" },
	{ "key": "shift+cmd+-",			"command": "editor.action.fontZoomOut", "when": "isMac" },
	{ "key": "cmd+-",				"command": "editor.action.fontZoomOut", "when": "isMac" },
	{ "key": "cmd+numpad0",			"command": "editor.action.fontZoomReset", "when": "isMac" },
	{ "key": "cmd+0",				"command": "editor.action.fontZoomReset", "when": "isMac" },
	{ "key": "ctrl+numpad_add",		"command": "editor.action.fontZoomIn", "when": "isWindows" },
	{ "key": "ctrl+shift+=",		"command": "editor.action.fontZoomIn", "when": "isWindows" },
	{ "key": "ctrl+=",				"command": "editor.action.fontZoomIn", "when": "isWindows" },
	{ "key": "ctrl+numpad_subtract","command": "editor.action.fontZoomOut", "when": "isWindows" },
	{ "key": "ctrl+shift+-",		"command": "editor.action.fontZoomOut", "when": "isWindows" },
	{ "key": "ctrl+-",				"command": "editor.action.fontZoomOut", "when": "isWindows" },
	{ "key": "ctrl+numpad0",		"command": "editor.action.fontZoomReset", "when": "isWindows" },
	{ "key": "ctrl+0",				"command": "editor.action.fontZoomReset", "when": "isWindows" },

```
